packages/@react-aria/grid/src/GridKeyboardDelegate.ts

```diff
@@ -137,6 +140,10 @@ export class GridKeyboardDelegate<T, C extends GridCollection<T>> implements Key
   }
 
   getKeyRightOf(key: Key) {
+    if (this.focusMode === 'row' && this.cycleMode === 'between') {
+      return this.getKeyBelow(key);
+    }
+
```

Comment:
> I'm keeping 'cycleMode' to help me nav between the different tags. Since each Tag is a different row, I need to add the functionality to nav between rows when using the left/right arrow keys.

`cycleMode` itself still feels confusing to me here. What exactly is the difference between `between`, `none`, and `within`? How would it be used in a grid component that doesn't only consist of rows laid out horizontally like TagGroup? Currently if feels very specific to TagGroup's implementation just because TagGroup is using our grid pattern but its rows are physically positioned horizontally rather than vertically. Now that we are no longer making focus move into the Tag itself we can probably simplify the logic required to support the left/right arrow moving focus between the Tags.

Some questions for discussion amongst the team:
- What should `cycleMode` do? Can we simplify it something called `horizontalMode` or something along the lines of that? The behavior I would then expect would be the following: if `horizontalMode` is true, have `getKeyRightOf` and `getKeyLeftOf` call `getKeyBelow` and `getKeyAbove` respectively and then we would get rid of the extra `cycleMode` behavior
- If we object to having `horizontalMode` in GridKeyboardDelegate, then perhaps TagGroup should have its own KeyboardDelegate? That way we redefine what `getKeyRightOf`/`getKeyLeftOf` return without modifying GridKeyboard delegate too much (and perhaps we could have the TagGroupKeyboardDelegate extend GridKeyboardDelegate if we need any of its stuff down the line). I personally prefer this approach unless we get further definition on what `cycleMode` should do for standard grid setups.

@tiffany-sy-lin Apologies about the lack of definition around this `cycleMode` behavior, perhaps we should do a quick design review with some of the team members to figure out where this `cycleMode` came from and what we want from this PR